src/Sulu/Bundle/TrashBundle/Infrastructure/Doctrine/Repository/TrashItemRepository.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TrashBundle\Infrastructure\Doctrine\Repository;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Sulu\Bundle\TrashBundle\Domain\Exception\TrashItemNotFoundException;
+use Sulu\Bundle\TrashBundle\Domain\Model\TrashItemInterface;
+use Sulu\Bundle\TrashBundle\Domain\Repository\TrashItemRepositoryInterface;
+
+final class TrashItemRepository implements TrashItemRepositoryInterface
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var EntityRepository<TrashItemInterface>
+     */
+    private $entityRepository;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+        $this->entityRepository = $entityManager->getRepository(TrashItemInterface::class);
+    }
+
+    public function addAndCommit(TrashItemInterface $trashItem): void
+    {
+        $this->entityManager->persist($trashItem);
+        $this->entityManager->flush();
+    }
+
+    public function removeAndCommit(TrashItemInterface $trashItem): void
+    {
+        $this->entityManager->remove($trashItem);
+        $this->entityManager->flush();
```

Comment:
should not be flushed here.

Reply:
This is only true, if the storage adapter is `doctrine`, but won't work with `elasticsearch` anymore

Reply:
We should in no case flush inside a repository or a handler as flushes are sideeffects.

In the doctrine implementation the repository should not flush if we implement ever a elasticsearch that is an own implementation which need to handle the flush therself.

Reply:
But the doctrine implementation and the elasticsearch implementation would only have a different repository, right? All other services should stay the same. So where should be "committed" instead?